backend/src/test/java/com/google/sps/FindMatchQueryTest.java

```diff
@@ -0,0 +1,278 @@
+  private ZonedDateTime TIME_1400ET;
+  private ZonedDateTime TIME_1450ET;
+  private ZonedDateTime TIME_1500ET;
+  private ZonedDateTime TIME_1501ET;
+  private ZonedDateTime TIME_1530ET;
+  private ZonedDateTime TIME_1600ET;
+  private ZonedDateTime TIME_1800ET;
+  private ZonedDateTime TIME_2000ET;
+  private ZonedDateTime TIME_1100PT;
+  private ZonedDateTime TIME_1200PT;
+  private ZonedDateTime TIME_1600PT;
+
+  private static final int DURATION_15_MINUTES = 15;
+  private static final int DURATION_30_MINUTES = 30;
+  private static final int DURATION_45_MINUTES = 45;
+  private static final int DURATION_60_MINUTES = 60;
+
+  private FindMatchQuery query;
+
+  @Before
+  public void setUp() {
+    // Set "current" date to  1/1/2020 2:00pm ET
+    ZonedDateTime currentDateTimeET =
+        ZonedDateTime.of(
+            /* year= */ 2020,
+            /* month= */ 1,
+            /* date= */ 1,
+            /* hour= */ 14,
+            /* minute= */ 0,
+            /* second= */ 0,
+            /* nanosecond= */ 0,
+            /* zone= */ ZoneId.of("US/Eastern"));
+    ZonedDateTime currentDateTimePT =
+        currentDateTimeET.withZoneSameInstant(ZoneId.of("US/Pacific"));
+
+    // Initialize time
+    TIME_1400ET = currentDateTimeET; // dummy start time
+    TIME_1450ET = getNewTimeToday(currentDateTimeET, 14, 50);
+    TIME_1500ET = getNewTimeToday(currentDateTimeET, 15, 0);
+    TIME_1501ET = getNewTimeToday(currentDateTimeET, 15, 1);
+    TIME_1530ET = getNewTimeToday(currentDateTimeET, 15, 30);
+    TIME_1600ET = getNewTimeToday(currentDateTimeET, 16, 0);
+    TIME_1800ET = getNewTimeToday(currentDateTimeET, 18, 0);
+    TIME_2000ET = getNewTimeToday(currentDateTimeET, 20, 0);
+    TIME_1100PT = getNewTimeToday(currentDateTimePT, 11, 0);
+    TIME_1200PT = getNewTimeToday(currentDateTimePT, 12, 0);
+    TIME_1600PT = getNewTimeToday(currentDateTimePT, 16, 0);
+
+    Clock clock = Clock.fixed(currentDateTimeET.toInstant(), currentDateTimeET.getZone());
+
+    query = new FindMatchQuery(clock);
+  }
+
+  @Test
+  public void compatibleTimeAndDuration() {
+    // Two participants who are compatible in available time AND duration
+    Participant participantA =
+        new Participant(
+            ID_DEFAULT, PERSON_A, TIME_1400ET, TIME_1600ET, DURATION_30_MINUTES, TIMESTAMP_DEFAULT);
+    Participant participantB =
+        new Participant(
+            ID_DEFAULT, PERSON_B, TIME_1400ET, TIME_1800ET, DURATION_15_MINUTES, TIMESTAMP_DEFAULT);
+
+    Match match = query.findMatch(Arrays.asList(participantA), participantB);
+
+    assertThat(match.getFirstParticipant().getUsername()).isEqualTo(PERSON_B);
+    assertThat(match.getSecondParticipant().getUsername()).isEqualTo(PERSON_A);
+    assertThat(match.getDuration()).isEqualTo(DURATION_15_MINUTES);
+  }
+
+  /** Return today's date with time of hour:minute */
+  public ZonedDateTime getNewTimeToday(ZonedDateTime dateTime, int hour, int minute) {
```

Comment:
nit: private static. That should also allow the initialization to move to the declaration, removing the corresponding setup from the setUp method

private static final ZonedDateTime TIME_1450ET = getNewTimeToday(currentDateTimeET, 14, 50);